src/realm/object-store/shared_realm.cpp

```diff
@@ -910,6 +910,22 @@ void Realm::close()
     m_coordinator = nullptr;
 }
 
+bool Realm::delete_files()
```

Comment:
I would very much agree so. The two main purposes of this function seem to be a general clean up and safety reasons.
In both cases I would expect `.backup.realm` to be removed too.
Is `.realm` relevant here too? This extension should be part of `path` in `Config` anyway right?
Are there other extensions we're missing and should add too? I've searched through the code base but couldn't find anything else that looked like it.

Reply:
Let's confirm with @finnschiermer if it is reasonable to delete the backups here, maybe this is exactly the case where a backup would come in handy?

I think we've covered all the extensions that I know of. This is why it would be nice to have them all listed in a nice self contained function :)